### PR TITLE
Enable Prettier for JavaScript files

### PR DESCRIPTION
**Pull request by kachkaev**
Follows #147 and #151.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -6,11 +6,11 @@
   "main": "index.js",
   "scripts": {
     "fix": "npm-run-all --continue-on-error \"fix:*\"",
-    "fix:markdownlint": "markdownlint --fix \"**/*\"",
+    "fix:markdownlint": "markdownlint --dot --fix \"**/*\"",
     "fix:prettier": "prettier --write \"**/*\"",
     "fix:yarn-deduplicate": "yarn install && yarn-deduplicate --strategy=fewer && yarn install",
     "lint": "npm-run-all --continue-on-error \"lint:*\"",
-    "lint:markdownlint": "markdownlint \"**/*\"",
+    "lint:markdownlint": "markdownlint --dot \"**/*\"",
```
Comment (kachkaev) [inline package.json:13]: Bonus: add `--dot` to `markdownlint` calls

```
markdownlint --help

  Options:
  ...
  -d, --dot                                   include files/folders with a dot (for example `.github`)
  ...
```

We’ll need this for `.github/pull_request_template.md`. I should have added this option in #147, but I did not know the option was necessary.

File: .prettierignore [hunk 2]
```diff
@@ -7,15 +7,23 @@
...
+!*.ts
+!*.tsx
 !*.yml
 # TODO: add more types
 
 # Keep license files as is
 LICENSE.md
 
+# Skip autogenerated files tracked by git
+/packages/engine/src/worker/runner/javascript/hash_stdlib.js
```
Comment (kachkaev) [inline .prettierignore:25]: @Alfred-Mountfield this file was obviously generated, so I excluded it from the Prettier list.

Screenshot 2022-01-12 at 13 07 39

I wondered if we could remove the file from git, but I could not find how it got generated. Never mind for now I guess.

Are there any other JS files we don’t want to cover by Prettier?

Comment (Alfred-Mountfield) [reply .prettierignore:25]: Copying in @litvand

**Review by Alfred-Mountfield — APPROVED** [review]
Looks good. Nice to get some JS linting in, it's been on our backlog for a while

File: packages/engine/stdlib/src/ts/neighbor.spec.ts [hunk 3]
```diff
@@ -25,19 +28,25 @@ test("find neighbors with same position", () => {
...
 
 test("find neighbors within a max radius of 4 and min radius of 3", () => {
   expect(
-    neighborsInRadius(ng, [na, nb, nc, nd, nf, nh, ni, nj], 4, 3.5)
+    neighborsInRadius(ng, [na, nb, nc, nd, nf, nh, ni, nj], 4, 3.5),
   ).toEqual([ni]);
 
   expect(
-    neighborsInRadius(ng, [na, nb, nc, nd, nf, nh, ni, nj], 7, 3.5, "euclidean", true)
+    neighborsInRadius(
```
Comment (Alfred-Mountfield) [inline packages/engine/stdlib/src/ts/neighbor.spec.ts:42]: @litvand Your favourite kind of inconsistent formatting haha

File: .prettierignore (hunk 2 above)
Comment (litvand) [reply .prettierignore:25]: `bundle_arrow.js` can also be skipped. Nothing else comes to mind.

Comment (Alfred-Mountfield) [reply .prettierignore:25]: Good point, probably worth reverting changes to the file to avoid such a big diff

**Review by Alfred-Mountfield — CHANGES_REQUESTED** [review]
@litvand made a good point about excluding `packages/engine/src/worker/runner/javascript/bundle_arrow.js`

File: packages/engine/stdlib/src/ts/neighbor.spec.ts (hunk 3 above)
Comment (litvand) [reply packages/engine/stdlib/src/ts/neighbor.spec.ts:42]: Yeah... Maybe `[na, nb, nc, nd, nf, nh, ni, nj]` can be assigned to a variable to make it consistent.

File: .prettierignore (hunk 2 above)
Comment (kachkaev) [reply .prettierignore]: ✅ 85221008d3dd385e53351ed57e187be5dab13aac

**Review by Alfred-Mountfield — APPROVED** [review]

**Pull request merged**